qiita_db/analysis.py

```diff
@@ -165,6 +169,71 @@ def create(cls, owner, name, description, parent=None, from_default=False):
         a_id = conn_handler.execute_queue(queue)[0]
         return cls(a_id)
 
+    @classmethod
+    def delete(cls, _id):
```

Comment:
I happy adding the check and the delete of those rows ... shouldn't be a problem.